wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -231,382 +246,483 @@ fn print_coin_amount(chain_config: &ChainConfig, value: Amount) -> String {
     value.into_fixedpoint_str(chain_config.coin_decimals())
 }
 
-pub async fn handle_wallet_command(
-    chain_config: &Arc<ChainConfig>,
-    rpc_client: &NodeRpcClient,
-    controller_opt: &mut Option<RpcController>,
-    command: WalletCommand,
-) -> Result<ConsoleCommand, WalletCliError> {
-    match command {
-        WalletCommand::CreateWallet {
-            wallet_path,
-            mnemonic,
-        } => {
-            utils::ensure!(
-                controller_opt.is_none(),
-                WalletCliError::WalletFileAlreadyOpen
-            );
-
-            // TODO: Support other languages
-            let language = wallet::wallet::Language::English;
-            let need_mnemonic_backup = mnemonic.is_none();
-            let mnemonic = match &mnemonic {
-                Some(mnemonic) => wallet_controller::mnemonic::parse_mnemonic(language, mnemonic)
-                    .map_err(WalletCliError::InvalidMnemonic)?,
-                None => wallet_controller::mnemonic::generate_new_mnemonic(language),
-            };
-
-            let wallet = RpcController::create_wallet(
-                Arc::clone(chain_config),
-                wallet_path,
-                mnemonic.clone(),
-                None,
-            )
-            .map_err(WalletCliError::Controller)?;
-
-            *controller_opt = Some(RpcController::new(
-                Arc::clone(chain_config),
-                rpc_client.clone(),
-                wallet,
-            ));
-
-            let msg = if need_mnemonic_backup {
-                format!(
-                    "New wallet created successfully\nYour mnemonic: {}\nPlease write it somewhere safe to be able to restore your wallet."
-                , mnemonic)
-            } else {
-                "New wallet created successfully".to_owned()
-            };
-            Ok(ConsoleCommand::Print(msg))
-        }
+pub struct CommandHandler {
+    state: Option<(U31, usize)>,
+}
```

Comment:
Would appreciate some comment describing what the tuple components are and what it means for the `Option` to be `None`.